ismatec/driver.py

```diff
@@ -1,50 +1,52 @@
-"""A single Ismatec Reglo ICC multi-channel peristaltic pump class."""
+"""Ismatec Reglo ICC multi-channel peristaltic pump driver.
+
+Distributed under the GNU General Public License v3
+Copyright (C) 2022 NuMat Technologies
+"""
 import logging
 
-from .util import Communicator, Protocol, SerialCommunicator, SocketCommunicator
+from .util import Protocol, SerialCommunicator, SocketCommunicator
 
 logger = logging.getLogger('ismatec')
 
 
 class Pump(Protocol):
-    """
-    Class for representing a single Ismatec Reglo ICC multi-channel peristaltic pump.
+    """Driver for a single Ismatec Reglo ICC peristaltic pump.
 
-    It can be controlled over a serial server (gateway) or direct serial.
+    The driver supports both serial and ethernet communication.
 
-    The, which can be controlled independently, are available as self.channels.
+    If the pump has multiple tube channels, they can be controlled
+    independently. See `self.channels` for available channels.
     """
 
-    def __init__(self, address=None, debug=False, **kwargs) -> None:
-        if debug:
-            logger.setLevel(logging.DEBUG)
-        """Initialize the Communicator and setup the pump to accept commands."""
-        # make a hardware Communicator object
-        if type(address) == str:
-            # serial
-            self.hw: Communicator = SerialCommunicator(address=address, **kwargs)
-        elif type(address) == tuple and len(address) == 2:
-            # socket
-            self.hw = SocketCommunicator(address=address, **kwargs)
+    def __init__(self, address='/dev/ttyUSB0', **kwargs) -> None:
+        """Connect to the pump and set initial state.
+
+        This optionally takes an address of the form
+            tcp://<<address>>:<<port>>
+        to talk to connected serial-to-ethernet converters.
+
+        Note that initialization sends multiple requests to the pump:
+        1. Enable independent channel addressing (TODO why?)
+        2. Disable asynchronous messages (TODO why?)
+        3. Get list of all available tube channels. (TODO why?)
+        4. Set initial running states (TODO why? to what?)
```

Comment:
The init calls a whole bunch of functions, and it won't be clear to you in two years why it calls these things and what the benefit is. Might as well dump the side effects and reasoning into the docstring for future you to read?

Reply:
Oh, they need to be cut out, but it needs testing with a real pump.  I should probably write an issue for that.